script/ModuleRegistry.sol

```diff
@@ -0,0 +1,400 @@
+import {LM_PC_Bounties_v1} from "@lm/LM_PC_Bounties_v1.sol";
+import {LM_PC_RecurringPayments_v1} from "@lm/LM_PC_RecurringPayments_v1.sol";
+import {LM_PC_Staking_v1} from "@lm/LM_PC_Staking_v1.sol";
+import {LM_PC_KPIRewarder_v1} from "@lm/LM_PC_KPIRewarder_v1.sol";
+import {LM_PC_PaymentRouter_v1} from "@lm/LM_PC_PaymentRouter_v1.sol";
+
+// Import scripts:
+import {ModuleFactory_v1} from "src/factories/ModuleFactory_v1.sol";
+import {OrchestratorFactory_v1} from "src/factories/OrchestratorFactory_v1.sol";
+
+//--------------------------------------------------------------------------
+// General Module Registry Information
+//--------------------------------------------------------------------------
+// Thif file acts as a general registry for all modules in the inverter protocol.
+//
+// It can be divided into two sections:
+//      1. A generalized 'implementation deployment' function
+//      2. A list of module implentation addresses and metadata to be used by inheriting scripts.
+//
+// For each Module, the list should declare:
+//      address moduleImplementation;  // The implementation address owill be stored here
+//      IModule_v1.Metadata moduleMetadata = IModule_v1.Metadata(
+//          1, 1, "https://github.com/inverter/module", "ModuleName"
+//      );
+//--------------------------------------------------------------------------
+
+contract ModuleRegistry is Script {
+    using Strings for string;
+    // ------------------------------------------------------------------------
+    // Fetch Environment Variables
+
+    uint deployerPrivateKey = vm.envUint("ORCHESTRATOR_ADMIN_PRIVATE_KEY");
+    address deployer = vm.addr(deployerPrivateKey);
+
+    function deployImplementation(
+        string memory contractName,
+        bytes memory constructorArgs
+    ) public returns (address implementation) {
+        vm.startBroadcast(deployerPrivateKey);
+        {
+            // Deploy the Module Implementation.
+
+            implementation = giantSwitchFromHell(contractName, constructorArgs);
+        }
+
+        vm.stopBroadcast();
+
+        // Log the deployed Module contract address.
+        console2.log(
+            "Deployment of %s Implementation at address %s",
+            contractName,
+            implementation
+        );
+    }
+
+    // @FHieser Before you start thinking about making an enum ;) You cannot easily print the contract name if you are donig that, which makes the logs a lot messier
+    function giantSwitchFromHell(
+        string memory contractName,
+        bytes memory constructorArgs
+    ) internal returns (address) {
+        // Orchestrator
+        if (Strings.equal(contractName, "Orchestrator_v1")) {
+            address forwarder = abi.decode(constructorArgs, (address));
+            return address(new Orchestrator_v1(forwarder));
+        }
+        // Protocol Contracts
+        else if (Strings.equal(contractName, "Governor_v1")) {
+            return address(new Governor_v1());
```

Comment:
We could even go with another if statement here in the Protocol Contracts:
If not created (address==0) create
If already created (address!=0) use existing